src/main/java/moze_intel/projecte/gameObjs/container/slots/transmutation/SlotInput.java

```diff
@@ -51,7 +52,9 @@ public void putStack(@Nonnull ItemStack stack)
 			}
 		}
 
-		inv.handleKnowledge(stack.copy());
+		if (EMCHelper.doesItemHaveEmc(stack)) {
+			inv.handleKnowledge(stack.copy());
```

Comment:
what's the if needed for, doesn't the isItemValid change guarantee anything coming in to putStack have a emc value?

Reply:
actually no.... It is an assumption that because something is an IItemEmc that it also has an EMC value, but it doesn't have to. For example, my EMC storage cells are IItemEmc's, but have no EMC value of their own.